mlir/lib/Dialect/Rock/utility/transformMapUtils.cpp

```diff
@@ -346,9 +348,249 @@ propagateUnmergeVectorization(T &&dimAndLength,
   return result;
 }
 
+// Key data structures for vectorization analysis
+using MergePairKey = std::pair<TransformMapAttr, TransformAttr>;
+
+using DimToMergeMap =
```

Comment:
For some of these, there might be an argument for SmallDenseMap (not sure I remember the spelling right)